### PR TITLE
Proxy status: fetching status only from running istiod pods + always returning status if present

### PR DESCRIPTION
**Pull request by xeviknal**
fixes https://github.com/kiali/kiali/issues/3605

1. The proxy-status is only fetch from pods in the `Running` state.
2. Kiali returns the proxy-status even if there is one non-responding `istiod` (following the `istioctl` approach). The proxy-status then might be partial because each `istiod` pod controls different set of envoys (e.g. `istiod1` controls `details` and `productpage`, `istiod2` controls `ratings` and `reviews`). If this scenario happens, then the Istio Component Status modal in the mast-head will be in red. Hence, it'd mean that the mesh is unstable and the admin/op needs to act.
3. New logs has been added to better troubleshoot this scenarios.

### CODE REVIEW

**Review by hhovsepy — APPROVED** [review]
Regression test passed.

File: kubernetes/istio.go [hunk 1]
```diff
@@ -214,26 +215,64 @@ func (in *K8SClient) GetProxyStatus() ([]*ProxyStatus, error) {
 		return nil, errors.New("unable to find any Pilot instances")
 	}
 
+	wg := sync.WaitGroup{}
```
Comment (lucasponce) [inline kubernetes/istio.go:218]: Note, I've fetched this PR locally and I guess I missed some review in the past.
This method is always fetching Pods from the wire even if that info is cached.

I think the business layer should combine and pass the pods to this method
.

**Review by lucasponce — APPROVED** [review]

**Comment by lucasponce** [comment]
It looks ok.
I propose some refactoring on that area, to pass the Pods to the kubernetes package from the business side when those pods are cached.
Otherwise, Kiali is bypassing caching for this logic.

But I'm fine if this refactoring is done in a followup PR.

File: kubernetes/istio.go (hunk 1 above)
Comment (xeviknal) [reply kubernetes/istio.go:218]: I don't see why we should use the pods from the cache. Using them may lead to scenarios where kiali fetches proxy status from old `istiod` that are either dead or in a not running phase. I think that the most reliable way to perform this operation is to use fresh `istiod` that at this moment are healthy.
What scenario am I missing?

Comment (lucasponce) [reply kubernetes/istio.go:218]: Well, it was more to follow same pattern of usage.
But it's an optimization, I understand that when proxy is cached, queries hit the cache for high level info and this fetch to the raw pods happens when building the new proxy status from the istiod status.

**Pull request merged**